hydra_base/lib/project.py

```diff
@@ -552,3 +589,22 @@ def clone_project(project_id,
     db.DBSession.flush()
 
     return new_project.id
+
+
+def _get_project_hierarchy(project_id, user_id):
+
+    proj = JSONObject(_get_project(project_id, user_id=user_id))
+
+    project_hierarchy = [proj]
+    if proj.parent_id:
```

Comment:
I think it would be better as:
```python
if hasattr(proj, 'parent_id') and proj.parent_id is not None:
```

Reply:
Walrus + getattr combines the present test and the not-None test...
```python
if parent_id := getattr(proj, "parent_id", None):
    ...
```

i.e. `parent_id` ends up as `None` in both cases and the `if` is not entered.

Reply:
Lets use this from now on.